fms/modules/attention.py

```diff
@@ -15,6 +15,57 @@
 from fms.modules.positions import PositionEncoder
 from fms.modules.tp import TPModule
 
+from fms.utils.gptq import GPTQConfig, custom_linear_repr
+
+try:
+    from auto_gptq.utils.import_utils import dynamically_import_QuantLinear
+    IS_AUTOGPTQ_AVAILABLE=True
+except:
+    IS_AUTOGPTQ_AVAILABLE=False
+
+
+def get_linear(
+    in_features: int,
+    out_features: int,
+    bias: bool,
+    gptq_config: Optional[GPTQConfig] = None,
```

Comment:
to make it more generic, we can make this an Enum that for this PR is "GPTQ" and "Linear", and we can add FP8 later

Reply:
and then have an extra_config param that we can pass as a dictionary that can hold the config for either FP8 or GPTQ

Reply:
maybe more like `linear_config` as a better name